From f8859a42bf98008b6fb8cda8d0338c800905e9ee Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Wed, 25 Oct 2017 10:36:05 +0300
Subject: [PATCH 1265/1345] crypto: inside-secure: remove unneeded memset

commit  acfc5e209ea0f49e901d30284e05dc1478710fba from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

the zeroing of a request is unneeded since we use the cryptoAPI
to set all request structure fields

Change-Id: I07421cd0eae524c699adb9358724336ffd5796b7
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45575
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/51646
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/inside-secure/cipher.c |    2 --
 drivers/crypto/inside-secure/hash.c   |    2 --
 2 files changed, 4 deletions(-)

diff --git a/drivers/crypto/inside-secure/cipher.c b/drivers/crypto/inside-secure/cipher.c
index 8735b8d..9485b3c 100644
--- a/drivers/crypto/inside-secure/cipher.c
+++ b/drivers/crypto/inside-secure/cipher.c
@@ -431,8 +431,6 @@ static int safexcel_cipher_exit_inv(struct crypto_tfm *tfm)
 	int ring = ctx->base.ring;
 	int ret;
 
-	memset(req, 0, sizeof(struct ablkcipher_request));
-
 	/* create invalidation request */
 	init_completion(&result.completion);
 	ablkcipher_request_set_callback(req, CRYPTO_TFM_REQ_MAY_BACKLOG,
diff --git a/drivers/crypto/inside-secure/hash.c b/drivers/crypto/inside-secure/hash.c
index f4e6753..cc7e1a0 100644
--- a/drivers/crypto/inside-secure/hash.c
+++ b/drivers/crypto/inside-secure/hash.c
@@ -452,8 +452,6 @@ static int safexcel_ahash_exit_inv(struct crypto_tfm *tfm)
 	int ring = ctx->base.ring;
 	int ret;
 
-	memset(req, 0, sizeof(struct ahash_request));
-
 	/* create invalidation request */
 	init_completion(&result.completion);
 	ahash_request_set_callback(req, CRYPTO_TFM_REQ_MAY_BACKLOG,
-- 
1.7.9.5

